packages/schematics/angular/module/index.ts

```diff
@@ -68,6 +70,74 @@ function addDeclarationToNgModule(options: ModuleOptions): Rule {
   };
 }
 
+function addRouteDeclarationToNgModule(
+  options: ModuleOptions,
+  project: WorkspaceProject,
+  routingModulePath: Path | undefined,
+): Rule {
+  return (host: Tree) => {
+    if (!options.route) {
+      return host;
+    }
+    if (options.route && !options.module) {
+      throw new Error('Module option required when creating a lazy loaded routing module.');
+    }
+
+    let path: string;
+    if (routingModulePath) {
+      path = routingModulePath as string;
+    } else {
+      path = options.module as string;
+    }
+
+    const text = host.read(path);
+    if (!text) {
+      throw new Error(`Couldn't find the module nor its routing module.`);
+    }
+
+    const ivyEnabled = isProjectUsingIvy(host, project);
+    const sourceText = text.toString('utf-8');
+    const addDeclaration = addRouteDeclarationToModule(
+      ts.createSourceFile(path, sourceText, ts.ScriptTarget.Latest, true),
+      path,
+      buildRoute(options, ivyEnabled),
+    ) as InsertChange;
+
+    const recorder = host.beginUpdate(path);
+    recorder.insertLeft(addDeclaration.pos, addDeclaration.toAdd);
+    host.commitUpdate(recorder);
+
+    return host;
+  };
+}
+
+function getRoutingModulePath(host: Tree, options: ModuleOptions): Path | undefined {
```

Comment:
This doesn't seem right. This method will always fall in the `catch` because you cannot locate a routing module with `findModuleFromOptions`

https://github.com/angular/angular-cli/blob/master/packages/schematics/angular/utility/find-module.ts#L110

Reply:
Actually, it should work. Frankly, I haven't looked into the function itself (`findModuleFromOptions`) but if it always used to hit the `catch`, [this test](https://github.com/angular/angular-cli/blob/389f9c7af55063f51d7a78e16af7ab163315346d/packages/schematics/angular/module/index_spec.ts#L148) would always fail.

In other words, we won't be able to access `app-routing.module.ts` and insert the declaration in that case.